lib/cc/config/prepare.rb

```diff
@@ -0,0 +1,68 @@
+module CC
+  module Config
+    class Prepare
+      class NoPrepareNeeded
+        def fetch
+          []
+        end
+      end
+
+      attr_reader :fetch
+
+      def self.from_yaml(data)
+        if data.present?
+          fetch = Fetch.from_yaml(data.fetch("fetch", []))
+
+          new(fetch: fetch)
+        else
+          NoPrepareNeeded.new
+        end
+      end
+
+      def initialize(fetch:)
+        @fetch = fetch
+      end
+
+      class Fetch
+        def self.from_yaml(data)
+          new(data.map { |d| Entry.from_yaml(d) })
+        end
+
+        def initialize(entries)
+          @entries = entries
+        end
+
+        def each(&block)
+          entries.each(&block)
+        end
+
+        private
+
+        attr_reader :entries
+
+        class Entry
+          attr_reader :url, :path
+
+          def self.from_yaml(data)
+            case data
+            when String then new(data)
+            when Hash then new(data.fetch("url"), data["path"])
+            end
+          end
+
+          def initialize(url, path = nil)
+            @url = url
+            @path = path || url.split("/").last
```

Comment:
We did more strenuous validation of the path being safe for security reasons in the old impl. I think we want to keep that.

https://github.com/codeclimate/codeclimate-yaml/blob/master/lib/cc/yaml/nodes/fetch.rb#L51

Less critical, but the new impl here also doesn't prevent path ending up as `""` in several ways. I assume that will blow up at `File.write`, but it may be worth validating earlier.